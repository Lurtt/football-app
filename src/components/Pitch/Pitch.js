import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { Row, Column, Player } from '../'

const PitchStyle = styled.div`
  margin: auto;
  background-color: #27ae60;
`

const generateColumns = (player, index) => (
  <Column key={index} maxWidth="320px" textAlign="center">
    <Player player={player} />
  </Column>
)

const generateRows = (row, index) => (
  <Row key={index} justifyContent="center">
    {row.map(generateColumns)}
  </Row>
)

export default function Pitch({ team }) {
  return <PitchStyle>{team && team.map(generateRows)}</PitchStyle>
}

Pitch.propTypes = {
  team: PropTypes.array.isRequired,
}
