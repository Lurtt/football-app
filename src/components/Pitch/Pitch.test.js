import React from 'react'
import renderer from 'react-test-renderer'

import Pitch from './'

jest.mock('../Loading/three-dots.svg', () => 'SVG_CONTENT_HERE')
jest.mock('../Player', () => 'Player component')

const team = [
  [
    {
      name: 'Hugo Lloris',
      formation_place: 1,
      type: 'Goalkeeper',
      position: 'GK',
    },
    {
      name: 'Serge Aurier',
      formation_place: 2,
      type: 'Defender',
      position: 'RB',
    },
  ],
]

describe('<Pitch />', () => {
  test('Should render', () => {
    const result = renderer.create(<Pitch team={team} />).toJSON()
    expect(result).toMatchSnapshot()
  })
})
