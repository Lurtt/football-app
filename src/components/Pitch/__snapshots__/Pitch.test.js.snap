// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pitch /> Should render 1`] = `
<div
  className="sc-ifAKCX gvVUmM"
>
  <div
    className="sc-bxivhb geHUii"
  >
    <div
      className="sc-bwzfXH lmmTFA"
    >
      <Player component
        player={
          Object {
            "formation_place": 1,
            "name": "Hugo Lloris",
            "position": "GK",
            "type": "Goalkeeper",
          }
        }
      />
    </div>
    <div
      className="sc-bwzfXH lmmTFA"
    >
      <Player component
        player={
          Object {
            "formation_place": 2,
            "name": "Serge Aurier",
            "position": "RB",
            "type": "Defender",
          }
        }
      />
    </div>
  </div>
</div>
`;
